test/Npgsql.Tests/NpgsqlParameterTests.cs

```diff
@@ -379,50 +340,49 @@ public void Parameter_null()
     }
 
     [Test]
-    [Ignore("")]
     public void Parameter_type()
     {
         NpgsqlParameter p;
 
         // If Type is not set, then type is inferred from the value
         // assigned. The Type should be inferred everytime Value is assigned
-        // If value is null or DBNull, then the current Type should be reset to Text.
-        p = new NpgsqlParameter();
+        // If value is null or DBNull, then the current Type should be reset to Unknown (DbType.Object and NpgsqlDbType.Unknown).
+        p = new NpgsqlParameter { Value = "" };
         Assert.That(p.DbType, Is.EqualTo(DbType.String), "#A1");
         Assert.That(p.NpgsqlDbType, Is.EqualTo(NpgsqlDbType.Text), "#A2");
         p.Value = DBNull.Value;
-        Assert.That(p.DbType, Is.EqualTo(DbType.String), "#B1");
-        Assert.That(p.NpgsqlDbType, Is.EqualTo(NpgsqlDbType.Text), "#B2");
+        Assert.That(p.DbType, Is.EqualTo(DbType.Object), "#B1");
+        Assert.That(p.NpgsqlDbType, Is.EqualTo(NpgsqlDbType.Unknown), "#B2");
         p.Value = 1;
         Assert.That(p.DbType, Is.EqualTo(DbType.Int32), "#C1");
         Assert.That(p.NpgsqlDbType, Is.EqualTo(NpgsqlDbType.Integer), "#C2");
         p.Value = DBNull.Value;
-        Assert.That(p.DbType, Is.EqualTo(DbType.String), "#D1");
```

Comment:
Uh, were we really defaulting to String/Text for null? I'm assuming this is some very ancient version of Npgsql (this test has been ignored forever), rather than anything recent...